exporters/otlp/profiles/src/main/java/io/opentelemetry/exporter/otlp/data/ProfileData.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.data;
+
+import io.opentelemetry.api.common.Attributes;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Represents a complete profile, including sample types, samples, mappings to binaries, locations,
+ * functions, string table, and additional metadata.
+ *
+ * @see "pprofextended.proto::Profile"
+ */
+@Immutable
+public interface ProfileData {
+
+  /** A description of the samples associated with each Sample.value. */
+  List<ValueTypeData> getSampleTypes();
+
+  /** The set of samples recorded in this profile. */
+  List<SampleData> getSamples();
+
+  /**
+   * Mapping from address ranges to the image/binary/library mapped into that address range.
+   * mapping[0] will be the main binary.
+   */
+  List<MappingData> getMappings();
+
+  /** Locations referenced by samples via location_indices. */
+  List<LocationData> getLocations();
+
+  /** Array of locations referenced by samples. */
+  List<Long> getLocationIndices();
+
+  /** Functions referenced by locations. */
+  List<FunctionData> getFunctions();
+
+  /** Lookup table for attributes. */
+  Attributes getAttributes();
```

Comment:
This breaks from the lookup table style we see elsewhere. Samples, Mappings, and Locations won't be able to reference this by index. Feels icky, but probably needs to be something like `List<Map.Entry<AttributeKey<?>,?>>`, or some other equivalent key/value pair holder.

Reply:
Yeah, I was going to revisit that when we get to the Marshalers. Right now I think it does kind of work, because the Attributes can be iterated in order. However, the existing attributes API doesn't really guarantee that characteristic, it's essentially a side-effect of the present implementation.  One option is to change Attributes to guarantee it, or to provide e.g. an asArray() type method that does, but otherwise yes it will have to be handled in the ProfileData API instead.
I think it's going to be a bigger headache for any user-facing API, where users will expect to be able to use Attributes more consistently. Hence having Attributes support ordering seems perhaps a better long term bet?

Reply:
> However, the existing attributes API doesn't really guarantee that characteristic, it's essentially a side-effect of the present implementation.

Yeah that's the problem. No guarantee.

> I think it's going to be a bigger headache for any user-facing API, where users will expect to be able to use Attributes more consistently.

The access patterns for attributes currently allow you to get by key, or iterate over all. Iterating over all with `Attributes.forEach` reflects the common way that serializer need to interact with attributes, where order typically doesnt matter. `Attributes.forEach` allows for iteration without allocation. If order does matter, `Attributes.asMap` is an escape hatch.

> One option is to change Attributes to guarantee it, or to provide e.g. an asArray() type method that does

I dont think we can guarantee it. Its possible to provide your own implementation of the `Attributes` interface. Guaranteeing the ordering would be retroactively adding a requirement to those implementations and is probably a violation of compatibility guarantees. The `asArray` option is akin to my suggestion of adding an accessor for `List<Map.Entry<AttributeKey<?>,?>>`. If its a problem unique to profiling, we might consider adding a helper `AttributeKeyValuePair` class only to the profiling signal. Not sure, but we can experiment with different things and see what works.